be/src/exec/pipeline/fragment_executor.cpp

```diff
@@ -255,7 +255,8 @@ void FragmentExecutor::_convert_data_sink_to_operator(PipelineBuilderContext* co
                                                                       result_sink->get_output_exprs(), _fragment_ctx);
         // Add result sink operator to last pipeline
         _fragment_ctx->pipelines().back()->add_op_factory(op);
-
+        // ResultSink's dop must set to 1
+        _fragment_ctx->pipelines().back()->source_operator_factory()->set_degree_of_parallelism(1);
```

Comment:
Why `ResultSinkOperator`'s do must be 1? Multiple `ResultSinkOperator` instances can  share on `BufferControlBlock`, because `BufferControlBlock::try_add_batch` has locked:

```C++
StatusOr<bool> BufferControlBlock::try_add_batch(TFetchDataResult* result) {
    std::unique_lock<std::mutex> l(_lock);

    if (_is_cancelled) {
        return Status::Cancelled("Cancelled BufferControlBlock::add_batch");
    }

    int num_rows = result->result_batch.rows.size();

    if ((!_batch_queue.empty() && (num_rows + _buffer_rows) > _buffer_limit) && !_is_cancelled) {
        return false;
    }

    if (_waiting_rpc.empty()) {
        _buffer_rows += num_rows;
        _batch_queue.push_back(result);
        _data_arriaval.notify_one();
    } else {
        auto* ctx = _waiting_rpc.front();
        _waiting_rpc.pop_front();
        ctx->on_data(result, _packet_num);
        delete result;
        _packet_num++;
    }
    return true;
}
```

Reply:
Ok